md-ui/src/main/java/com/wipro/ats/bdre/md/ui/PageController.java

```diff
@@ -46,10 +49,19 @@ public String welcome(@PathVariable("page") String page) {
 
     @RequestMapping(value = "/workflow/{pid}.page", method = RequestMethod.GET)
     @ResponseBody
-    public String getWorkflowDot(@PathVariable("pid") String pid) {
-        List<ProcessInfo> processInfos = new GetProcess().execute(new String[]{PARENTPROCESSID, pid});
-        Workflow workflow = new WorkflowPrinter().execute(processInfos, WORKFLOWCON + pid);
-        return workflow.getDot().toString();
+    public String getWorkflowDot(@PathVariable("pid") String pid, java.security.Principal principal) {
+        Workflow workflow=new Workflow();
+        try{
+            LOGGER.info("username is "+principal.getName());
+            List<ProcessInfo> processInfos = new GetProcess().execute(new String[]{PARENTPROCESSID, pid,"--username",principal.getName()});
+             workflow = new WorkflowPrinter().execute(processInfos, WORKFLOWCON + pid);
+            return workflow.getDot().toString();
+        }
+        catch (SecurityException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1166)